### PR TITLE
Fix `default.tensor` for `PauliRot`, `GlobalPhase`, and `Identity`

### PR DESCRIPTION
**Pull request by astralcai**
**Context:**
- `apply_operation` in `default.tensor` raises an error for `GlobalPhase` and `Identity` on no wires.
- `apply_operation_core_paulirot` in `default.tensor` does not account for the case where the `PauliRot` acts on a single wire, resulting in an error. 

**Description of the Change:**
- Add `apply_operation_core_global_phase` and handling for `Identity` to `default_tensor.py`
- Add logic to `apply_operation_core_paulirot` for the single-site case

**Related GitHub Issues:**
Fixes https://github.com/PennyLaneAI/pennylane/issues/6415
[sc-76302]

### CODE REVIEW

**Review by JerryChen97 — APPROVED** [review]
Thanks a lot @astralcai ! This is a very good catch; basically the MPO representations are not always guaranteed by the loop over Op str and the single-site case need special treatment. 
Also I believe for any TN related implementation we need a systematic tech notes in the future but that's far beyond current scope.

File: pennylane/devices/default_tensor.py [hunk 1]
```diff
@@ -949,14 +956,77 @@ def apply_operation_core_multirz(ops: qml.MultiRZ, device):
...
+
+    Parameters:
+        ops (qml.PauliRot): The operation containing the Pauli rotation parameters and qubits involved.
+        device: The quantum device that contains the state vector and settings for bond dimensions and cutoff values.
+
+    Returns:
+        None: The function modifies the quantum circuit's state directly.
+
+    Notes:
+        This implementation adheres to the MPO formalism for efficient computation within the Quimb framework,
```
Comment (astralcai) [inline pennylane/devices/default_tensor.py:1013]: Thanks @JerryChen97 for the detailed explanation, although I'm not sure if the explanation of implementation details need to be this extensive. @PietropaoloFrisoni what do you think?

Comment (JerryChen97) [reply pennylane/devices/default_tensor.py:1013]: Agree this is too much for a internal helper func. Let's stick to my first suggestion: only reminding the fact that in quimb def first two axes are the virtual legs and the last two axes are physical legs

**Review by PietropaoloFrisoni — APPROVED** [review]
LGTM!

**Pull request merged**